frame/elections-phragmen/src/lib.rs

```diff
@@ -406,55 +532,97 @@ decl_module! {
 		/// outcomes exist:
 		/// - `origin` is a candidate and not elected in any set. In this case, the bond is
 		///   unreserved, returned and origin is removed as a candidate.
-		/// - `origin` is a current runner up. In this case, the bond is unreserved, returned and
-		///   origin is removed as a runner.
+		/// - `origin` is a current runner-up. In this case, the bond is unreserved, returned and
+		///   origin is removed as a runner-up.
 		/// - `origin` is a current member. In this case, the bond is unreserved and origin is
 		///   removed as a member, consequently not being a candidate for the next round anymore.
 		///   Similar to [`remove_voter`], if replacement runners exists, they are immediately used.
-		#[weight = (2_000_000_000, DispatchClass::Operational)]
-		fn renounce_candidacy(origin) {
+		/// <weight>
+		/// If a candidate is renouncing:
+		/// 	Base weight: 16.57us
+		/// 	Complexity of candidate_count: 0.235
+		/// 	State reads:
+		/// 		- Candidates
+		/// 		- AccountBalance(who) (unreserve)
+		/// 	State writes:
+		/// 		- Candidates
+		/// 		- AccountBalance(who) (unreserve)
+		/// If member is renouncing:
+		/// 	Base weight: 46.25
+		/// 	State reads:
+		/// 		- Members, RunnersUp (remove_and_replace_member),
+		/// 		- AccountData(who) (unreserve)
+		/// 	State writes:
+		/// 		- Members, RunnersUp (remove_and_replace_member),
+		/// 		- AccountData(who) (unreserve)
+		/// If runner is renouncing:
+		/// 	Base weight: 46.25
+		/// 	State reads:
+		/// 		- RunnersUp (remove_and_replace_member),
+		/// 		- AccountData(who) (unreserve)
+		/// 	State writes:
+		/// 		- RunnersUp (remove_and_replace_member),
+		/// 		- AccountData(who) (unreserve)
+		///
+		/// TODO: and calls into ChangeMembers??
```

Comment:
@apopiak perhaps you can help here. Should this be noted? or I should wait for you PR for now?

Reply:
These definitely need to be factored in.
For Polkadot we know that it will be using `collective` under the hood, so you can cross-validate benchmark results for this function with [the annotations in the `collective` implementation here](https://github.com/paritytech/substrate/pull/5802/files#diff-fd8e3211907a8723b36fbf8504230548R771-R787).

Reply:
Your benchmarks should tell you all you need to know. The collective docs are just for cross-checking.

Reply:
As discussed offline, this will be addressed later.